lib/LedgerSMB/Admin/Command.pm

```diff
@@ -0,0 +1,144 @@
+
+package LedgerSMB::Admin::Command;
+
+=head1 NAME
+
+LedgerSMB::Admin::Command - ledgersmb-admin abstract command class
+
+=cut
+
+use strict;
+use warnings;
+
+use Log::Log4perl;
+use Pod::Find qw(pod_where);
+use Pod::Usage qw(pod2usage);
+
+use Moose;
+use namespace::autoclean;
+
+use LedgerSMB::Database;
+
+
+has config => (is => 'ro', required => 1,
+               isa => 'LedgerSMB::Admin::Configuration');
+has db => (is => 'rw');
+has logger => (is => 'ro', lazy => 1, builder => '_build_logger');
+
+sub _build_logger {
+    return Log::Log4perl->get_logger(ref $_[0]);
+}
+
+
+sub help {
+    my $self = shift;
+
+    pod2usage(-verbose => 99, -output => \*STDOUT,
+              -sections => 'SYNOPSIS|DESCRIPTION|COMMANDS|OPTIONS',
+              -input => pod_where({ -inc => 1 }, ref $self));
+}
+
+sub _before_dispatch {
+    my ($self) = shift;
+    $self->db(LedgerSMB::Database->new(
+                  connect_data => $self->config->get('connect_data'),
+              ));
+    return @_;
+}
+
+sub dispatch {
+    my ($self, $command, @args) = @_;
+
+    $command //= 'help';
+    die "Unknown command '$command'"
+        unless $command =~ m/^([a-zA-Z]+|run)$/;
```

Comment:
Aren't we testing that more precisely below?

Reply:
Not completely the same condition, no. I've replaced the code with something more clear, as per our discussion in the chat.